### PR TITLE
fix(reporter): Improve layout of Reporter Header

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/CT-355

### User facing changelog

Improve responsive design of the reporter header for both runner and runner-ct. Also reduce some technical debt; runner-ct no longer defines a custom header.

This removes the visual grouping of some elements in the `<ReporterHeader>` which make it really difficult to respond to changes in size. It's a significant user facing change and should be reviewed carefully.

### Additional details

When resizing the reporter, the various button groups move around strangely. This PR attempts to simplify the structure and make everything wrap in a more natural way.

The entire reporter header probably needs a rethink; Cypress and the UI have evolved a lot since it was first conceived. 

Video showing responsive (look at the header of the reporter as it resizes):

Before:

https://user-images.githubusercontent.com/19196536/112270602-9a4b3c80-8cc5-11eb-8093-ab4c01dcec5a.mov

After:

https://user-images.githubusercontent.com/19196536/112270609-9ddec380-8cc5-11eb-8cf1-6a10e9ad86e5.mov

## Runner (before/after)

image

image

## RunnerCt (before/after)

image

image

### PR Tasks

- [ ] Has the original issue or this PR been tagged with a release in ZenHub?  TODO: find out how to do this - CT Jira not connected to ZenHub.
- [x] Have tests been added/updated?

### CODE REVIEW

File: packages/runner-ct/src/app/SpecList/SpecList.tsx [hunk 1]
```diff
@@ -238,6 +238,8 @@ const fuzzyTransform = (spec: Cypress.Cypress['spec'], indexes: number[]) => {
   }
```
Comment (lmiller1990) [inline packages/runner-ct/src/app/SpecList/SpecList.tsx:238]: The `runner-ct` changes are unrelated, just fixing a missing dependency on a `useEffect` hook

File: packages/runner-ct/src/app/RunnerCt.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ const RunnerCt = namedObserver('RunnerCt',
 
     const runSpec = React.useCallback((file: FileNode) => {
       setActiveIndex(0)
-      const selectedSpec = props.state.specs.find((spec) => spec.absolute.includes(file.relative))
```
Comment (lmiller1990) [inline packages/runner-ct/src/app/RunnerCt.tsx:94]: Unrelated but linter was complaining so I fixed it

**Review by agg23 — COMMENTED** [review]
Looks fine to me, but I'll let the TR team approve.

**Comment by flotwig** [comment]
I'm not totally sold on removing the groups - I think the groupings are a helpful visual cue. But maybe the groupings could disappear once the sidebar has been shrunk to whatever the breakpoint is that causes things to start wrapping.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
I don’t think I have a problem with removing the grouping. Everything still seems obvious to what its use was before.

It's not obvious in the header in general what are actions and what are not actions though. I think we had this discussion before. Like, the Tests and auto scroll and refresh are actions, but the rest are just indicators - and there’s no real way to realize this in the current design . But I don’t really think this problem was introduced in this PR.

After using the new design, I think I would prefer to have to stop/refresh button right aligned where it always is. I use this pretty frequently, so the positioning of the button feels cumbersome, it should be aligned to the corner to better click. 

Can we have the design below and it still collapse like in this design?

Screen Shot 2021-03-25 at 8 58 34 PM

**Pull request closed without merging**